tests/client/test_explain_error.py

```diff
@@ -0,0 +1,59 @@
+### Test giskard_client.py
+# Path: tests/core/test_giskard_client.py
+import pytest
+from giskard.client.giskard_client import GiskardClient, GiskardError
+from giskard.client.giskard_client import explain_error
+from requests import Response
+import requests_mock
+
+UNKNOWN_ERROR = "Unknown error"
+
+
+def test_init_giskard_client_with_wrong_api_key():
+    """
+    Test giskard client with wrong api key
+    """
+    url = "http://giskard-host:9000"
+    api_key = "<wrong api key>"
+    with requests_mock.Mocker() as mocked_requests:
+        mocked_requests.register_uri(requests_mock.GET, url, json={})
+        mocked_requests.register_uri(requests_mock.GET, f"{url}/public-api/ml-worker-connect", status_code=401)
+
+        with pytest.raises(Exception) as exc_info:
```

Comment:
I think it could be:

```suggestion
        with pytest.raises(GiskardError) as exc_info:
```

and we do not need isinstance in this single test case anymore.